src/uct/tcp/tcp_ep.c

```diff
@@ -1184,9 +1253,14 @@ uct_tcp_ep_am_sendv(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
                        ((iov_cnt > 2) ? iov[2].iov_base : NULL),
                        ((iov_cnt > 2) ? iov[2].iov_len  : 0));
 
-    iface->outstanding += ep->tx.length - ep->tx.offset;
+    if (ucs_likely(!uct_tcp_ep_ctx_buf_need_progress(&ep->tx))) {
+        ucs_assert(status == UCS_OK);
+        uct_tcp_ep_ctx_reset(&ep->tx);
+    } else {
+        uct_tcp_ep_mod_events(ep, UCS_EVENT_SET_EVWRITE, 0);
+    }
```

Comment:
seems duplication of logic in uct_tcp_ep_progress_data_tx() (and maybe other places as well), extract common function?

Reply:
done